combat/CombatDamage.cpp

```diff
@@ -0,0 +1,211 @@
+#include "CombatDamage.h"
+
+#include "../util/Logger.h"
+#include "../util/GameRules.h"
+#include "../universe/Condition.h"
+#include "../universe/Enums.h"
+#include "../universe/Fighter.h"
+#include "../universe/Ship.h"
+#include "../universe/ShipDesign.h"
+#include "../universe/ShipPart.h"
+#include "../universe/ValueRef.h"
+
+// This knows about combat bouts, fighter launches, combatTargets conditions,
+// damage calculation (shortrange, fighters, shields)
+
+namespace {
+    ScriptingContext CombatContextWithShipTarget(std::shared_ptr<const Ship> source, float shields = 0.0f) {
+        // FIXME find default enemy - i use the given design as that at least exists (but might be non-targetable because of stealth)
+        //ErrorLogger() << "DESIGN " << GetPredefinedShipDesignManager().GetDesignID("08a58b08-0929-496d-84fc-faa91424ca21");
+        //auto target = std::make_shared<Ship>(ALL_EMPIRES, GetPredefinedShipDesignManager().GetDesignID("SD_MARK_1"), "SP_HUMAN");
+        //auto target = std::make_shared<Fighter>();
+        // XXX DesignID valid?
+        auto target = std::make_shared<Ship>(ALL_EMPIRES, source->DesignID(), "SP_HUMAN");
+        // target needs to have an ID != -1 to be visible, stealth should be low enough
+        // structure must be higher than zero to be valid target
+        target->SetID(-1000000); // XXX magic number AutoresolveInfo.next_fighter_id starts at -1000001 counting down
+        target->GetMeter(MeterType::METER_STRUCTURE)->Set(100.0f, 100.0f);
+        target->GetMeter(MeterType::METER_MAX_STRUCTURE)->Set(100.0f, 100.0f);
+        // Shield value is used for structural damage estimation
+        target->GetMeter(MeterType::METER_SHIELD)->Set(shields,shields);
+        GetUniverse().SetEmpireObjectVisibility(source->Owner(), target->ID(), Visibility::VIS_FULL_VISIBILITY);
+        return ScriptingContext(source, target);
+    }
+
+    ScriptingContext CombatContextWithFighterTarget(std::shared_ptr<const Ship> source) {
+        auto target = std::make_shared<Fighter>();
+        target->SetID(-1000000); // XXX magic number AutoresolveInfo.next_fighter_id starts at -1000001 counting down
+        GetUniverse().SetEmpireObjectVisibility(source->Owner(), target->ID(), Visibility::VIS_FULL_VISIBILITY);
+        return ScriptingContext(source, target);
+    }
+
+    std::vector<float> WeaponDamageCalcImpl(std::shared_ptr<const Ship> ship, const ShipDesign* design,
```

Comment:
purpose of `max` parameter is unclear

Reply:
that are also remnants. design is now redundant, will only pass ship instance.

max is used to use max*meters instead of *meters.

Reply:
> max is used to use max*meters instead of *meters.

please make that clearer in the code